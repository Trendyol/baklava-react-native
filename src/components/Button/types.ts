export type ButtonVariantVariantTypes = 'primary' | 'secondary' | 'tertiary';

export type ButtonKindVariantTypes =
  | 'default'
  | 'neutral'
  | 'success'
  | 'danger';

export type ButtonSizeVariantTypes = 's' | 'm' | 'l';
