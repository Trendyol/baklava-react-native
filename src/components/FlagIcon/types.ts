import { flagIconList } from './list';

export type FlagIconNameType = (typeof flagIconList)[number];
