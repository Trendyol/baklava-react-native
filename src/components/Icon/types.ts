import { iconList } from './list';

export type IconNameType = typeof iconList[number];
