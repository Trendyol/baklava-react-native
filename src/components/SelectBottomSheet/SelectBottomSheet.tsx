import React from 'react';
import {
  FlatList,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import Box, { BoxProps } from '../Box/Box';
import BottomSheet, { BottomSheetProps } from '../BottomSheet/BottomSheet';
import Checkbox from '../Checkbox/Checkbox';
import RadioButton from '../RadioButton/RadioButton';
import { FlagIconNameType } from '../FlagIcon/types';

interface Option {
  id: string;
  label: string;
  flagIcon?: FlagIconNameType | null;
  disabled?: boolean;
}

interface SelectBottomSheetProps extends BottomSheetProps {
  options: Option[];
  type: 'checkbox' | 'radio';
  onSelect?: (selectedValues: string[]) => void;
  selectedOption?: string[];
  maxVisibleItems?: number;
  pb?: BoxProps['pb'];
  accessibilityLabel?: string;
  testID?: string;
}

const SelectBottomSheet: React.FC<SelectBottomSheetProps> = ({
  options,
  type,
  onSelect,
  selectedOption,
  maxVisibleItems = 6,
  pb,
  accessibilityLabel,
  testID,
  ...bottomSheetProps
}) => {
  const [selectedValues, setSelectedValues] = React.useState<string[]>(
    selectedOption ?? [],
  );

  const testProps = React.useMemo(() => {
    return {
      accessibilityLabel: accessibilityLabel ?? testID,
      testID: testID ?? accessibilityLabel,
    };
  }, [testID, accessibilityLabel]);

  React.useEffect(() => {
    if (Array.isArray(selectedOption)) {
      setSelectedValues(selectedOption);
    }
  }, [selectedOption]);

  const handleSelect = (value: string) => {
    if (type === 'checkbox') {
      setSelectedValues(prev =>
        prev.includes(value) ? prev.filter(v => v !== value) : [...prev, value],
      );
    } else {
      setSelectedValues([value]);
    }
  };

  React.useEffect(() => {
    if (onSelect) {
      onSelect(selectedValues);
    }
  }, [selectedValues, onSelect]);

  const renderItem = ({ item, index }: { item: Option; index: number }) => {
    return (
      <Box key={item.id}>
        <TouchableOpacity
          onPress={() => handleSelect(item.id)}
          disabled={item.disabled}
          {...testProps}
          testID={`touchable-selectBottomSheet-${item.id}`}>
          {type === 'checkbox' ? (
            <Checkbox
              testID={`checkbox-selectBottomSheet-${item.id}`}
              label={item.label}
              flagIcon={item.flagIcon}
              checked={selectedValues.includes(item.id)}
              onPress={() => handleSelect(item.id)}
              disabled={item.disabled}
            />
          ) : (
            <RadioButton
              testID={`radiobutton-selectBottomSheet-${item.id}`}
              label={item.label}
              flagIcon={item.flagIcon}
              selected={selectedValues.includes(item.id)}
              onPress={() => handleSelect(item.id)}
              disabled={item.disabled}
            />
          )}
        </TouchableOpacity>
        {index < options.length - 1 && (
          <Box height={1} backgroundColor="borderColor" marginVertical="m" />
        )}
      </Box>
    );
  };

  return (
    <BottomSheet {...bottomSheetProps} {...testProps}>
      <Box pb={pb}>
        <ScrollView
          testID={'selectBottomSheet'}
          contentContainerStyle={styles.flatListContainer}
          horizontal
          bounces={false}>
          <FlatList
            data={options}
            renderItem={renderItem}
            keyExtractor={item => item.id}
            nestedScrollEnabled
            initialNumToRender={maxVisibleItems}
            maxToRenderPerBatch={maxVisibleItems}
            style={{ maxHeight: maxVisibleItems * 50 - 15 }}
          />
        </ScrollView>
      </Box>
    </BottomSheet>
  );
};

export default SelectBottomSheet;

const styles = StyleSheet.create({
  flatListContainer: {
    flex: 1,
    width: 'auto',
  },
});
