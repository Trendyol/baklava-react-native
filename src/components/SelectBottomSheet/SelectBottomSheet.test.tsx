import React from 'react';
import { fireEvent, render } from '../../test-utils';
import SelectBottomSheet from './SelectBottomSheet';
import { FlagIconNameType } from '../FlagIcon/types';

describe('SelectBottomSheet', () => {
  const options = [
    { id: '1', label: 'Option 1' },
    { id: '2', label: 'Option 2' },
    { id: '3', label: 'Option 3' },
  ];

  const mockOnSelect = jest.fn();
  const defaultProps = {
    options,
    type: 'checkbox' as const,
    onSelect: mockOnSelect,
    visible: true,
    closeButtonLabel: 'Close',
    selectButtonLabel: 'Select',
    flagIcon: 'TR',
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', () => {
    // when
    const { getByTestId } = render(<SelectBottomSheet {...defaultProps} />);

    //then
    expect(getByTestId('selectBottomSheet')).toBeTruthy();
  });

  test('renders correctly with radio type', () => {
    // when
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} type="radio" />,
    );

    //then
    expect(getByTestId('selectBottomSheet')).toBeTruthy();
  });

  test('should render SelectBottomSheet when visible true', () => {
    // when
    const { toJSON, queryByTestId } = render(
      <SelectBottomSheet {...defaultProps} />,
    );

    // then
    expect(queryByTestId('selectBottomSheet')).not.toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render SelectBottomSheet when visible false', () => {
    // when
    const { toJSON, queryByTestId } = render(
      <SelectBottomSheet {...defaultProps} visible={false} />,
    );

    // then
    const modalWrapper = queryByTestId('selectBottomSheet');
    expect(modalWrapper).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  test('should not render SelectBottomSheet when visible is false', () => {
    // when
    const { toJSON, queryByTestId } = render(
      <SelectBottomSheet {...defaultProps} />,
    );

    // then
    expect(queryByTestId('modal-wrapper')).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  test('should initialize with correct selected values', () => {
    // when
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} selectedOption={['1', '2']} />,
    );

    // then
    expect(
      getByTestId('checkbox-selectBottomSheet-1').props.children[0][0].props
        .variant,
    ).toBe('checked');
    expect(
      getByTestId('checkbox-selectBottomSheet-2').props.children[0][0].props
        .variant,
    ).toBe('checked');
    expect(
      getByTestId('checkbox-selectBottomSheet-3').props.children[0][0].props
        .variant,
    ).toBe('unchecked');
  });

  test('should update selected values on checkbox select', () => {
    // when
    const { getByTestId } = render(<SelectBottomSheet {...defaultProps} />);
    const checkbox = getByTestId('checkbox-selectBottomSheet-1');
    fireEvent.press(checkbox);

    // then
    expect(
      getByTestId('checkbox-selectBottomSheet-1').props.children[0][0].props
        .variant,
    ).toBe('checked');
  });

  test('should update selected values on checkbox unselect', () => {
    // given
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} selectedOption={['1', '2']} />,
    );
    const checkbox = getByTestId('checkbox-selectBottomSheet-1');

    // when
    fireEvent.press(checkbox);

    // then
    expect(
      getByTestId('checkbox-selectBottomSheet-1').props.children[0][0].props
        .variant,
    ).toBe('unchecked');
    expect(mockOnSelect).toHaveBeenCalledWith(['2']);
  });

  test('should update selected values on radio button select', () => {
    // when
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} type="radio" />,
    );
    const radioButton = getByTestId('radiobutton-selectBottomSheet-2');
    fireEvent.press(radioButton);

    // then
    expect(mockOnSelect).toHaveBeenCalledWith(['2']);
  });

  test('should call onSelect with selected radio button value', () => {
    // when
    const { getByText } = render(
      <SelectBottomSheet {...defaultProps} type="radio" />,
    );
    fireEvent.press(getByText('Option 3'));

    // then
    expect(mockOnSelect).toHaveBeenCalledWith(['3']);
  });

  test('should render flag icon if provided', () => {
    // given
    const optionsWithFlag = [
      { id: '1', label: 'Option 1', flagIcon: 'TR' as FlagIconNameType },
    ];

    // when
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} options={optionsWithFlag} />,
    );

    // then
    expect(getByTestId('flag-icon').props.title).toBe('TR');
  });

  test('should call onSelect if provided', () => {
    // when
    const { getByTestId } = render(<SelectBottomSheet {...defaultProps} />);
    const checkbox = getByTestId('checkbox-selectBottomSheet-1');
    fireEvent.press(checkbox);

    // then
    expect(mockOnSelect).toHaveBeenCalledWith(['1']);
  });

  test('should not call onSelect if not provided', () => {
    // when
    const { getByTestId } = render(
      <SelectBottomSheet {...defaultProps} onSelect={undefined} />,
    );
    const checkbox = getByTestId('checkbox-selectBottomSheet-1');
    fireEvent.press(checkbox);

    // then
    expect(mockOnSelect).not.toHaveBeenCalled();
  });
});
