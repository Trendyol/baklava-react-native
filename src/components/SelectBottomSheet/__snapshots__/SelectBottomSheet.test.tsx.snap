// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBottomSheet should not render SelectBottomSheet when visible is false 1`] = `
<View
  accessibilityLabel="undefined-wrapper"
  accessible={true}
  style={
    [
      {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
  testID="undefined-wrapper"
>
  <View
    accessibilityLabel="undefined-overlay"
    collapsable={false}
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#273142",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="undefined-overlay"
  />
  <View
    accessible={false}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderRadius": 8,
          "padding": 16,
        },
        {
          "maxHeight": "90%",
        },
      ]
    }
  >
    <View
      accessible={false}
      style={
        [
          {
            "flexDirection": "column",
            "gap": 14,
            "marginBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={false}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessibilityLabel="undefined-header-close"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="undefined-header-close"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#273142",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "left",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
        <Text
          accessibilityLabel="undefined-header-label"
          accessible={true}
          numberOfLines={1}
          style={
            [
              {
                "color": "#273142",
                "fontFamily": "Rubik-Medium",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 16,
                "textAlign": "center",
              },
              {
                "width": "50%",
              },
            ]
          }
          testID="undefined-header-label"
        />
        <View
          accessibilityLabel="undefined-header-select"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="undefined-header-select"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#F27A1A",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "right",
                },
              ]
            }
          >
            Select
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        style={
          [
            {
              "backgroundColor": "#D5D9E1",
              "height": 1,
              "marginHorizontal": -16,
            },
          ]
        }
      />
    </View>
    <RCTScrollView
      accessibilityLabel="undefined-body"
      bounces={false}
      showsVerticalScrollIndicator={true}
      testID="undefined-body"
    >
      <View>
        <RCTScrollView
          bounces={false}
          contentContainerStyle={
            {
              "flex": 1,
              "width": "auto",
            }
          }
          horizontal={true}
          testID="selectBottomSheet"
        >
          <View>
            <RCTScrollView
              data={
                [
                  {
                    "id": "1",
                    "label": "Option 1",
                  },
                  {
                    "id": "2",
                    "label": "Option 2",
                  },
                  {
                    "id": "3",
                    "label": "Option 3",
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              initialNumToRender={6}
              keyExtractor={[Function]}
              maxToRenderPerBatch={6}
              nestedScrollEnabled={true}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              stickyHeaderIndices={[]}
              style={
                {
                  "maxHeight": 285,
                }
              }
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-1"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 1
                      </Text>
                    </View>
                    <View
                      accessible={false}
                      style={
                        [
                          {
                            "backgroundColor": "#D5D9E1",
                            "height": 1,
                            "marginVertical": 16,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-2"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 2
                      </Text>
                    </View>
                    <View
                      accessible={false}
                      style={
                        [
                          {
                            "backgroundColor": "#D5D9E1",
                            "height": 1,
                            "marginVertical": 16,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-3"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 3
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`SelectBottomSheet should render SelectBottomSheet when visible false 1`] = `null`;

exports[`SelectBottomSheet should render SelectBottomSheet when visible true 1`] = `
<View
  accessibilityLabel="undefined-wrapper"
  accessible={true}
  style={
    [
      {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
  testID="undefined-wrapper"
>
  <View
    accessibilityLabel="undefined-overlay"
    collapsable={false}
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#273142",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="undefined-overlay"
  />
  <View
    accessible={false}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderRadius": 8,
          "padding": 16,
        },
        {
          "maxHeight": "90%",
        },
      ]
    }
  >
    <View
      accessible={false}
      style={
        [
          {
            "flexDirection": "column",
            "gap": 14,
            "marginBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={false}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessibilityLabel="undefined-header-close"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="undefined-header-close"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#273142",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "left",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
        <Text
          accessibilityLabel="undefined-header-label"
          accessible={true}
          numberOfLines={1}
          style={
            [
              {
                "color": "#273142",
                "fontFamily": "Rubik-Medium",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 16,
                "textAlign": "center",
              },
              {
                "width": "50%",
              },
            ]
          }
          testID="undefined-header-label"
        />
        <View
          accessibilityLabel="undefined-header-select"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="undefined-header-select"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#F27A1A",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "right",
                },
              ]
            }
          >
            Select
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        style={
          [
            {
              "backgroundColor": "#D5D9E1",
              "height": 1,
              "marginHorizontal": -16,
            },
          ]
        }
      />
    </View>
    <RCTScrollView
      accessibilityLabel="undefined-body"
      bounces={false}
      showsVerticalScrollIndicator={true}
      testID="undefined-body"
    >
      <View>
        <RCTScrollView
          bounces={false}
          contentContainerStyle={
            {
              "flex": 1,
              "width": "auto",
            }
          }
          horizontal={true}
          testID="selectBottomSheet"
        >
          <View>
            <RCTScrollView
              data={
                [
                  {
                    "id": "1",
                    "label": "Option 1",
                  },
                  {
                    "id": "2",
                    "label": "Option 2",
                  },
                  {
                    "id": "3",
                    "label": "Option 3",
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              initialNumToRender={6}
              keyExtractor={[Function]}
              maxToRenderPerBatch={6}
              nestedScrollEnabled={true}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              stickyHeaderIndices={[]}
              style={
                {
                  "maxHeight": 285,
                }
              }
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-1"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 1
                      </Text>
                    </View>
                    <View
                      accessible={false}
                      style={
                        [
                          {
                            "backgroundColor": "#D5D9E1",
                            "height": 1,
                            "marginVertical": 16,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-2"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 2
                      </Text>
                    </View>
                    <View
                      accessible={false}
                      style={
                        [
                          {
                            "backgroundColor": "#D5D9E1",
                            "height": 1,
                            "marginVertical": 16,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={false}
                    style={
                      [
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "flex-start",
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                      testID="checkbox-selectBottomSheet-3"
                    >
                      <View
                        accessibilityLabel="tickbox"
                        accessible={true}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderColor": "#AFBBCA",
                                "borderRadius": 2,
                                "borderWidth": 1,
                                "height": 16,
                                "justifyContent": "center",
                                "width": 16,
                              },
                            ],
                          ]
                        }
                        testID="tickbox"
                      />
                      <Text
                        accessibilityLabel="label"
                        accessible={true}
                        style={
                          [
                            {
                              "color": "#273142",
                              "fontFamily": "Rubik-Regular",
                              "fontSize": 14,
                              "marginLeft": 8,
                            },
                          ]
                        }
                        testID="label"
                      >
                        Option 3
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
